crypto/src/base58.rs

```diff
@@ -18,9 +17,6 @@ pub enum FromBase58CheckError {
     /// The input is missing checksum.
     #[error("missing checksum")]
     MissingChecksum,
-    /// Data is too long
-    #[error("data too long")]
-    DataTooLong,
     #[error("mismatched data lenght: expected {expected}, actual {actual}")]
```

Comment:
```suggestion
    #[error("mismatched data length: expected {expected}, actual {actual}")]
```

While we're here, can we fix a typo?